Sources/TestingMacros/Support/DiagnosticMessage.swift

```diff
@@ -438,32 +438,106 @@ struct DiagnosticMessage: SwiftDiagnostics.DiagnosticMessage {
   /// number of arguments when applied to the given function declaration.
   ///
   /// - Parameters:
-  ///   - attribute: The `@Test` or `@Suite` attribute.
+  ///   - attribute: The `@Test` attribute.
   ///   - functionDecl: The declaration in question.
   ///
   /// - Returns: A diagnostic message.
   static func attributeArgumentCountIncorrect(_ attribute: AttributeSyntax, on functionDecl: FunctionDeclSyntax) -> Self {
     let expectedArgumentCount = functionDecl.signature.parameterClause.parameters.count
-    switch expectedArgumentCount {
-    case 0:
+    if expectedArgumentCount == 0 {
       return Self(
         syntax: Syntax(functionDecl),
         message: "Attribute \(_macroName(attribute)) cannot specify arguments when used with '\(functionDecl.completeName)' because it does not take any",
         severity: .error
       )
-    case 1:
+    } else {
       return Self(
         syntax: Syntax(functionDecl),
-        message: "Attribute \(_macroName(attribute)) must specify an argument when used with '\(functionDecl.completeName)'",
-        severity: .error
+        message: "Attribute \(_macroName(attribute)) must specify arguments when used with '\(functionDecl.completeName)'",
```

Comment:
We should add "function" to both messages so it reads like "Attribute 'Test' must specify arguments when used with **function** 'foo(x:)'". I think that's stylistically closer to what the compiler emits.